R/labeller.r

```diff
@@ -311,6 +311,8 @@ as_labeller <- function(x, default = label_value, multi_line = TRUE) {
       x(labels)
     } else if (is.function(x)) {
       default(lapply(labels, x))
+    } else if (is.formula(x)) {
+      default(lapply(labels, as_function(x)))
```

Comment:
Is there a reason why you pivoted on the earlier approach of converting x to a function once at the top?